lib/fastlane/plugin/wpmreleasetoolkit/actions/android/android_generate_po_file_from_metadata.rb

```diff
@@ -0,0 +1,136 @@
+require_relative '../../helper/po_extended'
+require 'fastlane/action'
+require 'gettext/po'
+
+module Fastlane
+  module Actions
+    class AndroidGeneratePoFileFromMetadataAction < Action
+
+      # SPECIAL_KEYS are keys that need to be treated specially
+      SPECIAL_KEYS = %w[release_notes release_notes_previous].freeze
+      REQUIRED_KEYS_TO_COMMENT_HASH = {
+        full_description: 'Multi-paragraph text used to display in the Play Store. Limit to 4000 characters including spaces and commas!',
+        title: 'Title to be displayed in the Play Store. Limit to 30 characters including spaces and commas!',
+        short_description: 'Short description of the app to be displayed in the Play Store. Limit to 80 characters including spaces and commas!',
+        release_notes_short: nil,
```

Comment:
[Tag `19.9` is one of those versions where we needed to include that release_notes_short alternative, where you can find inspiration for the associated comment for it](https://github.com/wordpress-mobile/WordPress-Android/blob/19.9/WordPress/metadata/PlayStoreStrings.po#L42)

Reply:
Just noticed how we can have two `release_notes_short` entries, one for `{VERSION}` and one for `{VERSION-1}`. 

That makes sense: if we need to provide `release_notes_{VERSION-1}` then also a `release_notes_short_{VERSION-1}` could be needed because `release_notes_{VERSION-1}` could have been  close to ≈350 characters.

Should we also make `release_notes_short_{VERSION-1}` also required then, just like `release_notes_previous` is required?

How about getting the `code_freeze` lanes to also create a `release_notes_short_{VERSION-1}.txt` (next to `release_notes_short_{VERSION}.txt` that' already a hard requirement?)

[I have just added support](https://github.com/Riziero/release-toolkit/commit/a8421fa0e3520f417d834534439503b1d9b3b7d3) for `release_notes_short`

Reply:
> > Should we also make release_notes_short_{VERSION-1} also required then, just like release_notes_previous is required?
> 
> Definitively no, on the contrary even.
> 
> As discussed in other comments, the `*_previous` entries should NOT be required, because we don't want the `.po` file generation to fail just because we don't have the `.txt` of the previous version (in theory we'd always have one, at least after the first 2 iterations on using the new action, given what the plan about creating that previous file by copying the old one at the moment of the `:code_freeze` lane… but that shouldn't be a _hard requirement_, because while we want it if we can, it's not dramatic and shouldn't block the rest of the release process if we don't have it).
> 
> In addition, `*_short_*` keys should **definitively NOT** be required keys either, since, as you saw it yourself given we didn't have those keys in the last couple of past releases, those are not always present (in particular we only provide them if the main release notes were >350 chars but don't provide ones otherwise)

I was completely off track, rushing things does not help. I wrote some proper garbage.

> So TL;DR:
> 
> 1. The `release_notes` key should be required, as we need to always have release notes for each new version we release
> 2. But the `release_notes_short` shouldn't because we don't always have/need one
> 3. The `release_notes_previous` one shouldn't be required either, because, while we expect to have it be created during `code_freeze` and in theory it will always be there, it shouldn't be a hard requirement that would stop and block the release process if it happens to be missing for whatever reason. At most, it could be a warning, but not a blocker.
> 4. As for `release_notes_short_previous`, it doubly-shouldn't be required for the combination of reasons 2 and 3.

Introducing `release_notes_short_previous` was more of a suggestion of something that could also be done in `code_freeze` - if it does exist at all at that point in time. I'll leave it for another time.

[With the following commit ](https://github.com/wordpress-mobile/release-toolkit/commit/00ee2c7e78034ab86b1e4621afddb9614dfb8c0b) I have now addressed these points:
- explicitly defined REQUIRED_KEYS (no longer figuring them off KEYS_TO_COMMENTS) as also suggested [below](https://github.com/wordpress-mobile/release-toolkit/pull/414#discussion_r991435242)
- `release_notes_previous.txt` and  `release_notes_short.txt are` no longer required, just `UI.important` if not present
- Broke the circular dependency by which tests would rely on the Actions' implementation
- Adjusted the tests accordingly
- Moved KNOWN_OPTIONAL_KEYS_TO_COMMENT_HASH to the `_helper` class

I would nice to add more tests and revamp them as suggested [below](https://github.com/wordpress-mobile/release-toolkit/pull/414#discussion_r991521304)

Reply:
All this long thread is resolved by:
-  a8421fa0e3520f417d834534439503b1d9b3b7d3 that introduced a default comment for `release_notes_short`
- 00ee2c7e78034ab86b1e4621afddb9614dfb8c0b that made `release_notes_short` and `release_notes_previous` not required
- a8421fa0e3520f417d834534439503b1d9b3b7d3 that made `release_notes_short` part of `SPECIAL_KEYS`